src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -1131,7 +1131,7 @@ internal static int GetEntry(CompactTree tree, Page page, ushort entriesOffset,
             encodedKey.Key.CopyTo(output.ToSpan());
 
             var outputSpan = output.ToSpan();
-            key = output[^1] == 0 ? outputSpan : outputSpan.Slice(0, outputSpan.Length - 1);
+            key = outputSpan[^1] == 0 ? outputSpan[0..^1] : outputSpan;
```

Comment:
AFAIR we sliced the buffer to delete the null character (`\0`).

Reply:
That is been done here to be consistent on usage.